lib/omise/res/OmiseApiResource.php

```diff
@@ -7,265 +7,307 @@
+     */
+    protected static function g_create($clazz, $url, $params, $publickey = null, $secretkey = null)
+    {
+        $resource = call_user_func(array($clazz, 'getInstance'), $clazz, $publickey, $secretkey);
+        $result   = $resource->execute($url, self::REQUEST_POST, $resource->getResourceKey(), $params);
+        $resource->refresh($result);
+
+        return $resource;
+    }
 
-    // Finally.
-    $request_url = dirname(__FILE__).'/../../../tests/fixtures/'.$request_url.'-'.strtolower($requestMethod).'.json';
+    /**
+     * Updates the resource with the given parameters in an associative array.
+     *
+     * @param  string $url
+     * @param  array  $params
+     *
+     * @throws Exception|OmiseException
+     */
+    protected function g_update($url, $params)
+    {
+        $result = $this->execute($url, self::REQUEST_PATCH, $this->getResourceKey(), $params);
+        $this->refresh($result);
+    }
 
-    // Make a request from Curl if json file was not exists.
-    if (!file_exists($request_url)) {
+    /**
+     * Destroys the resource.
+     *
+     * @param  string $url
+     *
+     * @throws Exception|OmiseException
+     *
+     * @return OmiseApiResource
+     */
+    protected function g_destroy($url)
+    {
+        $result = $this->execute($url, self::REQUEST_DELETE, $this->getResourceKey());
+        $this->refresh($result, true);
+    }
 
-      // Get a directory that's file should contain.
-      $request_dir = explode('/', $request_url);
-      unset($request_dir[count($request_dir) - 1]);
-      $request_dir = implode('/', $request_dir);
+    /**
+     * Reloads the resource with latest data.
+     *
+     * @param  string $url
+     *
+     * @throws Exception|OmiseException
+     */
+    protected function g_reload($url)
+    {
+        $result = $this->execute($url, self::REQUEST_GET, $this->getResourceKey());
+        $this->refresh($result);
+    }
 
-      // Create directory if it not exists.
-      if (!file_exists($request_dir)) {
-        mkdir($request_dir, 0777, true);
-      }
+    /**
+     * Makes a request and returns a decoded JSON data as an associative array.
+     *
+     * @param  string $url
+     * @param  string $requestMethod
+     * @param  array  $params
+     *
+     * @throws OmiseException
+     *
+     * @return array
+     */
+    protected function execute($url, $requestMethod, $key, $params = null)
+    {
+        // If this class is execute by phpunit > get test mode.
+        if (preg_match('/phpunit/', $_SERVER['SCRIPT_NAME'])) {
+            $result = $this->_executeTest($url, $requestMethod, $key, $params);
+        } else {
+            $result = $this->_executeCurl($url, $requestMethod, $key, $params);
+        }
+
+        // Decode the JSON response as an associative array.
+        $array = json_decode($result, true);
+
+        // If response is invalid or not a JSON.
+        if (count($array) === 0 || ! isset($array['object']))
```

Comment:
@nimid fine, this one not change much and no need to test/fix unit test

done 👍